File: Problem_2/problem_2.erl
-module(problem_2).
-compile(export_all).

start()->
    factorials(1, 1, 4000000, 0).

factorials(A, B, Max, Accum) when A + B > Max ->
    io:format("result: ~p~n", [Accum]);
factorials(A, B, Max, Accum) ->
    case (A + B) rem 2 of
	0 -> 
	    factorials(A + B, A, Max, Accum + A + B);
	1 ->
	    factorials(A + B, A, Max, Accum)
    end.

File: project-euler/Problem_02/problem_2.go
package main

import (
	"fmt"
)

func main() {
	result := 0
	x, y := 0, 1

	for (x < 4000000) {
		x, y = y, x + y
		if y % 2 == 0 {
			result += y			
		}		
	}

	fmt.Println("Result :", result)
	
}